test/test_use_cmake/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# This test checks that it is possible to use the
+# CMake artifacts generated by the build.
+
+cmake_minimum_required(VERSION 3.1.0)
+project(test_use)
+
+file(GENERATE
+  OUTPUT
+    "${CMAKE_CURRENT_BINARY_DIR}/main.cpp"
+  CONTENT
+    [=[
+#include <portaudio.h>
+int main()
+{
+  Pa_Initialize();
+  Pa_Terminate();
+}
+]=]
+)
+
+find_package(PortAudio REQUIRED)
```

Comment:
`find_package()` will fail because CMake will try to find `FindPortAudio.cmake` in `cmake\modules` which is not provided. If intent of the implementation was to find built PortAudio shared library and link it to the test executable then probably `find_library()` is the best way to go, or `FindPortAudio.cmake` has to be provided with `find_library()` for built PortAudio library.

Reply:
@dmitrykos - Do we simply need to replace "find_package" with "find_library"?
Or is it a more complicated change?

If so then we can make that simple change and merge it if it works.

Reply:
Previously I commented wrongly about `find_package()` because PA CMake build actually creates `PortAudioConfig.cmake` in the installation path (I did not know that it was done).

My earlier proposer with `find_library()` ignores linkage of 3-rd party deps (ALSA, Regex, Jack, ...) which are pulled by `PortAudioConfig.cmake`, so some workloads

> Do we simply need to replace "find_package" with "find_library"?

The change for a test can as follows starting from the line 21:

```cmake
find_library(PORTAUDIO_LIB portaudio)
find_path(PORTAUDIO_INCLUDE portaudio.h)
include_directories(${PORTAUDIO_INCLUDE})

add_library(test_use_static STATIC "${CMAKE_CURRENT_BINARY_DIR}/main.cpp")
add_library(test_use_shared SHARED "${CMAKE_CURRENT_BINARY_DIR}/main.cpp")
add_executable(test_use_exe "${CMAKE_CURRENT_BINARY_DIR}/main.cpp")

target_link_libraries(test_use_static PRIVATE ${PORTAUDIO_LIB})
target_link_libraries(test_use_shared PRIVATE ${PORTAUDIO_LIB})
target_link_libraries(test_use_exe PRIVATE ${PORTAUDIO_LIB})
```

I will experiment with possible fixes in my fork.

Reply:
@philburk after experimenting in my fork I managed to get idea of this PR working, see my post in this thread.